staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -310,6 +312,9 @@ func (s preparedGenericAPIServer) NonBlockingRun(stopCh <-chan struct{}) error {
 	go func() {
 		<-stopCh
 		close(internalStopCh)
+		if stoppedCh != nil {
+			<-stoppedCh
```

Comment:
do we want to log that we're waiting for requests to complete here? do we want to set a time limit on how long we'll wait?

Reply:
The shutdown of the server has a context already with the timeout. Following the api docs, we don't need another timeout here.

Reply:
ok